backend/src/main/java/meetup/connect/event/EventService.java

```diff
@@ -37,10 +42,38 @@ public EventDto getById(Long id) {
         .orElseThrow(() -> new MeetUpException(MeetUpError.EVENT_NOT_FOUND));
   }
 
-  public void deleteById(Long id) {
-    if (!eventRepository.existsById(id)) {
-      throw new MeetUpException(MeetUpError.EVENT_NOT_FOUND);
+  public void deleteById(Long id, String email) {
+    Event event =
+        eventRepository
+            .findById(id)
+            .orElseThrow(() -> new MeetUpException(MeetUpError.EVENT_NOT_FOUND));
+    User user = userService.findByEmail(email);
+    if (!isOwnerOfEvent(user, event.getOwner())) {
+      throw new MeetUpException(MeetUpError.INSUFFICIENT_PERMISSIONS);
     }
     eventRepository.deleteById(id);
   }
+
+  public void manageEventAttendance(String email, Long id) {
+    Event event =
+        eventRepository
+            .findById(id)
+            .orElseThrow(() -> new MeetUpException(MeetUpError.EVENT_NOT_FOUND));
+    User user = userService.findByEmail(email);
+    if (isAttendingEvent(user, event)) {
+
+      event.getAttendees().remove(user);
```

Comment:
suggestion: implement a method in the `Event` class, such as `Event::removeAttendee`, `Event::addAttendee`. It'll be easier to debug and maintain, you can get `NullPointerException` right now btw